### PR TITLE
Fix bucket creation issues with Minio chart

### LINKED ISSUE #3128: minio fails to create default bucket
**Is this a request for help?**:
yes
---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG REPORT

**Version of Helm and Kubernetes**:
Helm and tiller version 2.7.2. Kubernetes version 1.8.4.

**Which chart**:
minio

**What happened**:
The configured default bucket wasn't created. The software accessing it reports
`Error: com.amazonaws.services.s3.model.AmazonS3Exception The specified bucket does not exist (Service: Amazon S3; Status Code: 404; Error Code: NoSuchBucket; Request ID: 3L137; S3 Extended Request ID: 3L137)` and `mc ls myminio` fails to list any buckets. My chart which in turn depends on minio is called "storage", that's why the "storage-" prefixes below come from.

Even though the storage-minio-create-bucket pod is deleted after running, I was able to get logs from it while it was running using `while true; do kubectl logs -f storage-minio-create-bucket; done`, it outputs (some newlines added for readability):
```
Error from server (NotFound): pods "storage-minio-create-bucket" not found
...
Error from server (NotFound): pods "storage-minio-create-bucket" not found
Error from server (BadRequest): container "minio-mc" in pod "storage-minio-create-bucket" is waiting to start: ContainerCreating
failed to get container status {"" ""}: rpc error: code = OutOfRange desc = EOF
failed to get container status {"" ""}: rpc error: code = OutOfRange desc = EOF
failed to get container status {"" ""}: rpc error: code = OutOfRange desc = EOF
failed to get container status {"" ""}: rpc error: code = OutOfRange desc = EOF
failed to get container status {"" ""}: rpc error: code = OutOfRange desc = EOF
failed to get container status {"" ""}: rpc error: code = OutOfRange desc = EOF
failed to get container status {"" ""}: rpc error: code = OutOfRange desc = EOF
failed to get container status {"" ""}: rpc error: code = OutOfRange desc = EOF
failed to get container status {"" ""}: rpc error: code = OutOfRange desc = EOF
failed to get container status {"" ""}: rpc error: code = OutOfRange desc = EOF
failed to get container status {"" ""}: rpc error: code = OutOfRange desc = EOF
failed to get container status {"" ""}: rpc error: code = OutOfRange desc = EOF
failed to get container status {"" ""}: rpc error: code = OutOfRange desc = EOF
failed to get container status {"" ""}: rpc error: code = OutOfRange desc = EOF
Error from server (NotFound): pods "storage-minio-create-bucket" not found
...
```

Meanwhile tiller logged this:
```
[tiller] 2017/12/21 12:57:21 preparing install for storage
...
[tiller] 2017/12/21 12:57:22 performing install for storage
[tiller] 2017/12/21 12:57:22 executing 1 pre-install hooks for storage
[tiller] 2017/12/21 12:57:22 hooks complete for pre-install storage
...
[kube] 2017/12/21 12:57:23 creating 18 resource(s)
[kube] 2017/12/21 12:57:23 beginning wait for 18 resources with timeout of 5m0s
...
[kube] 2017/12/21 12:57:38 Pod is not ready: mynamespace/storage-memcached-0
[tiller] 2017/12/21 12:57:39 executing 1 post-install hooks for storage
[kube] 2017/12/21 12:57:39 building resources from manifest
[kube] 2017/12/21 12:57:40 creating 1 resource(s)
[kube] 2017/12/21 12:57:40 Watching for changes to Pod storage-minio-create-bucket with timeout of 5m0s
[kube] 2017/12/21 12:57:40 Add/Modify event for storage-minio-create-bucket: ADDED
[tiller] 2017/12/21 12:57:40 hooks complete for post-install storage
[tiller] 2017/12/21 12:57:40 deleting post-install hook storage-minio-create-bucket for release storage due to "hook-succeeded" policy
[kube] 2017/12/21 12:57:40 Starting delete for "storage-minio-create-bucket" Pod
[kube] 2017/12/21 12:57:40 Using reaper for deleting "storage-minio-create-bucket"
[storage] 2017/12/21 12:57:40 updating release "storage.v1"
...
[kube] 2017/12/21 12:57:41 Doing get for Secret: "storage-minio-user"
[kube] 2017/12/21 12:57:41 get relation pod of object: mynamespace/Secret/storage-minio-user
...
[kube] 2017/12/21 12:57:41 Doing get for PersistentVolume: "storage-minio"
[kube] 2017/12/21 12:57:41 get relation pod of object: /PersistentVolume/storage-minio
...
```

**What you expected to happen**:
When `speficying minio.defaultBucket.enabled=true` and `minio.defaultBucket.name=files`, the `files` bucket should be created.

**How to reproduce it** (as minimally and precisely as possible):
I created a chart which in turn requires the minio 0.4.2 chart. Its values.yaml contains:
```yaml
minio:
  mode: standalone
  accessKey: Secret...
  # On AWS, SecretAccessKey example has 41 chars
  secretKey: Secret...
  persistence:
    size: 256Mi
    storageClass: "storage-minio"
  serviceType: ClusterIP
  defaultBucket:
    enabled: true
    name: files
    policy: none
```

**Anything else we need to know**:
I suspect this might happen due to minio-create-bucket running too early before minio itself is fully started up and ready to create a bucket. Maybe https://github.com/kubernetes/helm/issues/3282 (helm post-ready hook feature proposal) or manually awaiting minio startup in the pod would help?

### PR DESCRIPTION
**Pull request by nitisht**
Also update to latest Minio release

Fixes #3128

### CODE REVIEW

File: stable/minio/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 description: Distributed object storage server built for cloud applications and devops.
 name: minio
-version: 0.4.3
+version: 0.4.4
```
Comment (mattfarina) [inline stable/minio/Chart.yaml]: This appears to be a breaking change (e.g., the logger section being removed from the `values.yaml` file). Shouldn't this be bumped to 0.5.0?

Comment (nitisht) [reply stable/minio/Chart.yaml]: agreed, updated the PR

**Comment by mattfarina** [comment]
@nitisht I've tested this a few time in minikube (k8s 1.9.0 installed via kubeadm) and the default bucked isn't created.

My config around the default bucket looks like...
```
defaultBucket:
  enabled: true
  ## If enabled, must be a string with length > 0
  name: bucket
  ## Can be one of none|download|upload|public
  policy: none
  ## Purge if bucket exists already
  purge: true
```

**Comment by nitisht** [comment]
ok let me check on Minikube, I tested on K8S cluster on GCS and it worked.

Do you have any logs from the bucket-creation-job ?

File: stable/minio/templates/post-install-create-bucket-pod.yaml [hunk 2]
```diff
@@ -1,41 +1,46 @@
...
-  name: {{ template "minio.fullname" . }}-create-bucket
+  name: {{ template "minio.fullname" . }}-make-bucket-pod
   labels:
     app: {{ template "minio.fullname" . }}
     chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
     release: "{{ .Release.Name }}"
     heritage: "{{ .Release.Service }}"
   annotations:
-    "helm.sh/hook": post-install
+    "helm.sh/hook": post-install,post-update
```
Comment (mattfarina) [inline stable/minio/templates/post-install-create-bucket-pod.yaml]: It's not `post-update`. It should be `post-upgrade`. The error in the hook was not causing it to fire for me.

Comment (nitisht) [reply stable/minio/templates/post-install-create-bucket-pod.yaml]: great catch @mattfarina , thanks! I have updated the PR

**Comment by mattfarina** [comment]
tested and seeing the bucket created now.

/lgtm

**Pull request merged**